onedal/_device_offload.py

```diff
@@ -153,11 +164,35 @@ def _get_host_inputs(*args, **kwargs):
     return q, hostargs, hostkwargs
 
 
-def _extract_usm_iface(*args, **kwargs):
+def _extract_array_attr(*args, **kwargs):
```

Comment:
The addition of the array_api and array_api_device returns show a lot of overlap with ```get_namespace```. Ideally the array_api functionality should be located in _array_api.py to make things easier for future developers. I would say DRY principles could reduce maintenance too.  _extract_array_attr function isn't used here, so I am not sure why extracting the array_api and array_api_device can't be done in _array_api.py instead.

Reply:
Outdated. please see discussion at [_extract_usm_iface](https://github.com/intel/scikit-learn-intelex/pull/1861#discussion_r1741263421)